### ISSUE TITLE
splice array method throws an error

### ISSUE THREAD

**Issue by codenamezjames** [cronvel/nested-proxies#1]
I have been messing around with it for quite a while and I can't seem to figure out how to fix this issue.

`TypeError: 'get' on proxy: property 'prototype' is a read-only and non-configurable data property on the proxy target but the proxy did not return its actual value (expected '[object Array]' but got '[object Object]')`

Here is the test i used to produce this error
```javascript
const DeepProxy = require('./lib/DeepProxy')

var array = [1, 2];
var proxy = new DeepProxy(array);

proxy.splice(0, 1);
```

**Comment by codenamezjames** [comment]
However, I was able to do this and it gave me the desired results.
```javascript
const DeepProxy = require('./lib/DeepProxy')

var array = [1, 2];
var proxy = new DeepProxy(array);

array.splice(0, 1);

console.log(proxy, array)
```

**Comment by cronvel** [comment]
@codenamezjames I will look into it once I get some times... It could be critical for some libs I use in production that depends on this package...

**Comment by cronvel** [comment]
Apparently, we can pass blindly the trap's arguments to its `Reflect` counterpart except for the `construct` handler, which was unexpected. 
And `.splice()` uses `new` to create the array of deleted items, so it triggers the bug, unlike most array methods...

@codenamezjames thanks for reporting! ;)

**Edit:** BTW It's fixed in v0.1.9.
